docs/metrics.md

```diff
@@ -0,0 +1,12 @@
+# Build Controller Metrics
+
+The Build component exposes several metrics to help you monitor the health and behavior of your build resources.
+
+Following build metrics are exposed at service `build-operator-metrics` on port `8383`.
+
+| Name | Type | Description | Label | Status |
+| ---- | ---- | ----------- | ----- | ------ |
+| `build_builds_total` | Counter | Number of total created Builds. | build_buildstrategy=<build_buildstrategy_name> | experimental |
+| `build_buildruns_total` | Counter | Number of total new triggered BuildRuns. | build_buildstrategy=<build_buildstrategy_name> | experimental |
+| `build_buildrun_establish_duration_seconds` | Histogram | BuildRun establish duration in seconds. | build_buildstrategy=<build_buildstrategy_name><br>buildrun_namespace=<buildrun_namespace> | experimental |
+| `build_buildrun_completion_duration_seconds` | Histogram | BuildRun completion duration in seconds. | build_buildstrategy=<build_buildstrategy_name><br>buildrun_namespace=<buildrun_namespace> | experimental |
```

Comment:
I suggest to have the namespace as label in all metrics. Also, I suggest shortening the labels to `namespace` and `buildstrategy`.

I'd like to see descriptions for the metrics that clearly describe what the metric measures. Currently only the last one fits the implementation:

- build_builds_total: the description is fine with me (I would remove the total out of it but that's only cosmetic). But, the implementation measures something completely different. It measures all creations and updates of a build resource that leads to a Build with Registered=True. That's something completely different. I comment this in the code again.
- build_buildruns_total: the description is unclear because the term _triggered_ is not defined in the context of this project. I suggest we measure what the name specifies = the number of created buildruns. The implementation currently measures the number of completed buildruns. I comment this in the code again.
- build_buildrun_establish_duration_seconds: the name and description are unclear because the term _established_ is not defined in the context of this project. I suggest to name it like `build_buildrun_created_to_running_duration_seconds` with a corresponding description `Duration in seconds of a build run from its creation to its start.`
- build_buildrun_completion_duration_seconds: the name and description are okay but not specific enough I think. I would name it `build_buildrun_created_to_completed_duration_seconds` with a description `Duration in seconds of a build run from its creation to its completion.`

Reply:
Yes, I think about two questions when I doing the development:
- **I suggest to have the namespace as label in all metrics.**
I used namespace as label for all metrics, but at that time, I also face the metric limitation problem. I think it is not good like tekton to export lots of metrics which make the monitoring system crash. So I reduce the build and buildrun count just for buildstrategy. I also would like to add more label like namespace. But not sure if it will generated lots of metrics when we use both buildstrategy and namespace for all custom metrics.

- **Also, I suggest shortening the labels to namespace and buildstrategy.**
I also used the short name before, but I found there are some labels in the system which may make the end user confusing: such as: namespace, namespace.name or kubernetes.namespace.name. 
But I agree we can change to short name if we have a good document.
<img width="244" alt="截屏2020-06-22 下午3 53 39" src="https://user-images.githubusercontent.com/18108042/85262782-d8020700-b4a0-11ea-9ba1-a207c4a8370d.png">



About the name and description, how about:
- build_builds_registered
- build_buildruns_completed
- `build_buildrun_created_to_running_duration_seconds` is too long, I think we can have a short name but have a better description
- Same for `build_buildrun_created_to_completed_duration_seconds`

Reply:
Changed to use:
- build_builds_registered_total
- build_buildruns_completed_total

And short the label to `buildstrategy` and `namespace`